6-prediction/05-logistic-regression/notes.qmd

```diff
@@ -139,57 +155,115 @@ We can visualize the approach that logistic regression takes by sketching the pr
+
+**False Positives**
+: Predicting a 1 that is in fact a 0
+
+**False Negatives**
+: Predicting a 0 that is in fact a 1
+
+### Misclassification rate
 
+While thinking about failing is not exciting, it does give us a way to think about how well our classification model is doing. We would like the number of misclassifications over a large number of e-mails to be as small as possible. This fraction can be quantified as the **misclassification rate** or $\text{MCR}$ for short.
 
-## Fitting Logistic Regression
+**Misclassification Rate**
 
-The computational machinery for fitting logistic regression looks almost identical to what we used for linear least squares regression. The primary function we'll use is `glm()`.
+$$\text{MCR} = \frac{\text{no. FP} + \text{no. FN}}{\text{no. of predictions}} = \frac{\text{no. of misclassifications}}{\text{no. of predictions}}$$
+
+Just like in a multiple linear regression context, we can fit a suite of different models on the *training set* and evaluate their performance on the *testing set*. Now, our model type of choice is logistic regression and our evaluation metric is $\text{MCR}$. 
+
+## The Ideas in Code
+
+### Fitting Logistic Regression with `glm()`
+
+The computational machinery for fitting logistic regression looks almost identical to what we used for linear least squares regression. The primary function we'll use is `glm()`. We will also use a different `broom` function called `tidy()` to display the coefficients $b_0$ and $b_1$. `tidy()` can also be used in a linear regression context as well.
 
 ```{r}
 #| code-fold: false
 
 m1 <- glm(spam ~ log(num_char), data = email, family = "binomial")
-m1
+m1 |>
+  tidy() |>
+  select(term,estimate,std.error)
 ```
 
 These coefficients are a bit more challenging to interpret since they're no longer linearly related to the response. The sign of the coefficient for `log(num_char)`, however, is informative: it tells us that messages with more characters will be predicted to have a lower probability of being spam.
 
-Let's take a look at the predictions that this model makes back into the data set that it was trained on (also called the fitted values). When using the `predict()` function on logistic regression models, there are several different types of predictions that it can return, so be sure to specify `type = "response"`.
+
+### Evaluating model performance
+
+#### Calculating $\text{MCR}$
+
+Let's take a look at the predictions that this model makes back into the data set that it was trained on. When using the `predict()` function on logistic regression models, there are several different types of predictions that it can return, so be sure to use the additional argument `type` and supply it with `"response"`. This will return $\hat{p}_i$ (probability of a `1`) for each observation.
 
 ```{r}
 #| code-fold: false
+p_hat_m1 <- predict(object = m1, 
```

Comment:
Would it be worth folding this with subsequent code into blocks that look like the block from the last PS? That’s to say: a series of mutates before the summarize?

Reply:
That's a good idea. I will approve, merge and publish right now, but this is worth adding an issue for.